tiledb/sm/consolidator/fragment_consolidator.cc

```diff
@@ -597,9 +597,10 @@ FragmentConsolidator::create_buffers(
       std::accumulate(buffer_weights.begin(), buffer_weights.end(), 0);
 
   // Allocate space for each buffer.
+  auto adjusted_budget = total_budget / total_weights * total_weights;
   for (unsigned i = 0; i < buffer_num; ++i) {
-    buffer_sizes[i] =
-        std::max<uint64_t>(1, total_budget * buffer_weights[i] / total_weights);
+    buffer_sizes[i] = std::max<uint64_t>(
+        1, adjusted_budget * buffer_weights[i] / total_weights);
```

Comment:
Possibly also consider truncation here, might the values ever 'close enough' that the division loses a fraction that would be needed for 'large enough' buffer?

Reply:
With the adjusted budget, no. As said above, adjusted_budget / total_weights will always be a whole number.